src/app/profile/offline-meditation/offline-meditation.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component } from '@angular/core';
+import { MeditationService } from '../../meditation/meditation.service';
+import * as moment from 'moment';
+
+/**
+ * Component for logging offline meditations
+ */
+@Component({
+  selector: 'offline-meditation',
+  template: require('./offline-meditation.html'),
+  styles: [
+    require('./offline-meditation.css')
+  ]
+})
+export class OfflineMeditation {
+
+  walking: string = '';
+  sitting: string = '';
+  date: string = '';
+  time: string = '';
+  sending: boolean = false;
+
+  constructor(public meditationService: MeditationService) {}
+
+  checkDateTime() {
```

Comment:
I think this could be done easier with http://momentjs.com/docs/#/parsing/is-valid/.